app/src/main/java/org/dhis2/utils/customviews/ScanTextView.kt

```diff
@@ -0,0 +1,146 @@
+package org.dhis2.utils.customviews
+
+import android.Manifest
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.util.AttributeSet
+import android.view.View
+import android.widget.ImageView
+import androidx.core.content.ContextCompat
+import androidx.databinding.ViewDataBinding
+import com.google.android.material.textfield.TextInputEditText
+import com.google.android.material.textfield.TextInputLayout
+import kotlinx.android.synthetic.main.scan_text_view.view.delete
+import kotlinx.android.synthetic.main.scan_text_view.view.descIcon
+import org.dhis2.BR
+import org.dhis2.Bindings.Bindings
+import org.dhis2.R
+import org.dhis2.databinding.ScanTextViewAccentBinding
+import org.dhis2.databinding.ScanTextViewBinding
+import org.dhis2.usescases.qrScanner.QRActivity
+import org.hisp.dhis.android.core.common.ObjectStyle
+import org.hisp.dhis.android.core.common.ValueTypeRenderingType
+
+class ScanTextView : FieldLayout {
+
+    private lateinit var iconView: ImageView
+    private lateinit var editText: TextInputEditText
+    private lateinit var inputLayout: TextInputLayout
+    private lateinit var descriptionLabel: ImageView
+    private lateinit var binding: ViewDataBinding
+    private lateinit var onScanClick: OnScanClick
+    private lateinit var onScanResult: (String?) -> Unit
+
+    constructor(context: Context) : super(context) {
+        init(context)
+    }
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
+        init(context)
+    }
+
+    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(
+        context,
+        attrs,
+        defStyle
+    ) {
+        init(context)
+    }
```

Comment:
There is a easier way to achieve all of this in Kotlin with very few lines of code.

``` kotlin
class ScanTestView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyle: Int = 0
) : FieldLayout(context, attrs, defStyle) {

    private lateinit var iconView: ImageView
    ...
    private lateinit var onScanResult: (String?) -> Unit
    
    init {
        init(context)
    }
}
````

the `@JvmOverloads` will create all the different constructors depending on the defaults parameter of the main constructor. 

Then we use the `init` block to call the FieldLayout.init(context) method when initializing this class.